### PR TITLE
Cp1419 copy owner soft limits for new organization users

### PR DESCRIPTION
**Pull request by juanignaciosl**
This closes CartoDB/cartodb-platform#1419.

### CODE REVIEW

**Comment by juanignaciosl** [comment]
@gfiorav CR this, please.

File: app/controllers/admin/organization_users_controller.rb [hunk 1]
```diff
@@ -34,10 +40,22 @@ def edit
...
+    validation_failure = !soft_limits_validation(@user, params[:user], current_user.organization.owner)
+
+    @user.set_fields(
+      params[:user],
+      [
+        :username, :email, :password, :quota_in_bytes, :password_confirmation,
+        :twitter_datasource_enabled, :soft_geocoding_limit, :soft_here_isolines_limit])
     @user.organization = current_user.organization
-    @user.username = "#{@user.username}"
+    @user.username = @user.username
```
Comment (gfiorav) [inline app/controllers/admin/organization_users_controller.rb]: I'm sure there is a good reason but, why is this needed?

Comment (juanignaciosl) [inline app/controllers/admin/organization_users_controller.rb]: Not needed at all, deleted.

File: app/controllers/admin/organization_users_controller.rb [hunk 2]
```diff
@@ -34,10 +40,22 @@ def edit
...
     current_user.copy_account_features(@user)
+
+    raise Carto::UnprocesableEntityError.new("Soft limits validation error") if validation_failure
```
Comment (gfiorav) [inline app/controllers/admin/organization_users_controller.rb]: `validation_failure` is not reevaluated after :46 ?

File: config/initializers/error_notifier.rb [hunk 3]
```diff
@@ -17,7 +17,7 @@ module CartoDB
   # Extra can contain `:request` and `:user`
   # Deprecated because of that `extra` content limitation. Use `report_exception` instead.
   def self.notify_exception(e, extra={})
-    if Rails.env.development? || Rails.env.test?
+    if Rails.env.development?
```
Comment (gfiorav) [inline config/initializers/error_notifier.rb]: Interesting, today I was thinking that expecting on `notify_exception` in tests is very convenient. Better take the temptation away :smiling_imp:

Comment (gfiorav) [inline config/initializers/error_notifier.rb]: Also, maybe notify_error, notify_debug, etc should be treated equally (they still allow test env).

File: spec/requests/carto/api/organization_users_controller_spec.rb [hunk 4]
```diff
@@ -59,47 +124,97 @@
...
+      login(@organization.owner)
+      username = 'soft-limits-false-user'
+      params = user_params_soft_limits(username, false)
+
+      post api_v1_organization_users_create_url(name: @organization.name), params
+
+      last_response.status.should eq 200
+
+      @organization.reload
+      last_user_created = @organization.users.find { |user| user.username == username }
```
Comment (gfiorav) [inline spec/requests/carto/api/organization_users_controller_spec.rb]: Is the hound on to something here? Why not `detect`?

**Comment by gfiorav** [comment]
Apart from the minor comments, seems largely :package: to me !

File: spec/requests/carto/api/organization_users_controller_spec.rb (hunk 4 above)
Comment (juanignaciosl) [inline spec/requests/carto/api/organization_users_controller_spec.rb]: There was a debate but it's no longer true. `detect` is not more readable or fast, we should use `find`. There's a mismatch from my local rubocop to Hound, I'll try to disable this message in this PR. Just ignore it.

File: app/controllers/admin/organization_users_controller.rb (hunk 2 above)
Comment (juanignaciosl) [inline app/controllers/admin/organization_users_controller.rb]: No, it's not, and it shouldn't be. I've tried to explain that in the comment. I need to evaluate it with both the "old" @user and the params.

File: config/initializers/error_notifier.rb (hunk 3 above)
Comment (juanignaciosl) [inline config/initializers/error_notifier.rb]: I wouldn't expect on `notify_exception`. Nevertheless this change doesn't prevent you to do it.

You're right about other methods. Removing logging from all of them.

File: spec/requests/carto/api/organization_users_controller_spec.rb (hunk 4 above)
Comment (gfiorav) [inline spec/requests/carto/api/organization_users_controller_spec.rb]: ok!

File: config/initializers/error_notifier.rb (hunk 3 above)
Comment (gfiorav) [inline config/initializers/error_notifier.rb]: Wouldn't it? Test run in test env right? This check won't pass no?

**Comment by gfiorav** [comment]
I've added some more comments but still :+1: to me!

File: config/initializers/error_notifier.rb (hunk 3 above)
Comment (juanignaciosl) [inline config/initializers/error_notifier.rb]: If you stub the _call_ to `notify_exception` it doesn't matter what's inside the method.

Comment (gfiorav) [inline config/initializers/error_notifier.rb]: Ah! I was talking about expecting on some specific logging output. Thanks for the explanation anyways!
